Bottom: bdd79cf0331634ab1059166f6ab1dacf10e6f822
Top:    5a7662d466053913a4bfe767f8aeb41e48e282dd
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-17 12:35:20 +0800

autofs-5.0.7 - use ulimit max open files if greater than internal maximum

When setting the maximum number of allowed file handles the current setting
should be checked before setting it. If the ulimit command has been used to
increase the maximum to larger than what automount would ask for then honour
it.


---

diff --git a/daemon/automount.c b/daemon/automount.c
index 3f9337f..2bbffb0 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -2105,8 +2105,11 @@ int main(int argc, char *argv[])
 		exit(1);
 	}
 
-	rlim.rlim_cur = MAX_OPEN_FILES;
-	rlim.rlim_max = MAX_OPEN_FILES;
+	res = getrlimit(RLIMIT_NOFILE, &rlim);
+	if (res == -1 || rlim.rlim_max <= MAX_OPEN_FILES)  {
+		rlim.rlim_cur = MAX_OPEN_FILES;
+		rlim.rlim_max = MAX_OPEN_FILES;
+	}
 	res = setrlimit(RLIMIT_NOFILE, &rlim);
 	if (res)
 		printf("%s: can't increase open file limit - continuing",
